src/ray/core_worker/core_worker.cc

```diff
@@ -2202,19 +2202,26 @@ void CoreWorker::HandleGetObjectLocationsOwner(
     return;
   }
   auto object_id = ObjectID::FromBinary(request.object_id());
-  absl::optional<absl::flat_hash_set<NodeID>> node_ids =
-      reference_counter_->GetObjectLocations(object_id);
-  Status status;
-  if (node_ids.has_value()) {
-    for (const auto &node_id : node_ids.value()) {
-      reply->add_node_ids(node_id.Binary());
-    }
-    status = Status::OK();
-  } else {
-    status = Status::ObjectNotFound("Object " + object_id.Hex() + " not found");
+  auto callback = [this, object_id, reply, send_reply_callback](
+                      const absl::flat_hash_set<NodeID> &locations, int64_t object_size,
+                      int64_t current_version) {
+    io_service_.post([object_id, reply, send_reply_callback, locations, object_size,
```

Comment:
Why do we need io_service_.post? Can we just reply?

Reply:
My thinking was that we don't want to block whichever thread is updating the reference state with constructing `n` protos and replying to `n` RPCs, and I've seen elsewhere that we tend to offload such callbacks to the IO event loop. Are you saying that constructing and sending the RPC response will be pretty lightweight and that posting to this to the event loop isn't necessary?

Reply:
+1 let's just respond directly. Posting to the IO loop adds significant overheads.